### PR TITLE
Filter truncated quant.sf files

### PR DESCRIPTION
**Pull request by wvauclain**
## Issue Number

#2191 

## Purpose/Implementation Notes

Since the quant.sf files that were causing issues are the ones that are missing their headers, let's filter out files that are missing the correct header.

## Types of changes

What types of changes does your code introduce?

- Bugfix (non-breaking change which fixes an issue)

## Functional tests

I added a new quant test file that is one of the truncated files from our human quantpendia and modified the quant.sf only smash test to make sure it was filtered.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots

Please attach any screenshots that illustrate these changes.

### CODE REVIEW

File: workers/data_refinery_workers/processors/smashing_utils.py [hunk 1]
```diff
@@ -1026,16 +1026,39 @@ def write_tsv_json(job_context):
...
+    parallelize downloading quantsf files. """
+    (sample, latest_computed_file, output_file_path) = download_tuple
     try:
         latest_computed_file.get_synced_file_path(path=output_file_path)
-    except:
+    except Exception as e:
         # Let's not fail if there's an error syncing one of the quant.sf files
-        logger.exception("Failed to sync computed file", computed_file_id=latest_computed_file.pk)
+        logger.exception(
+            "Failed to sync computed file", computed_file_id=latest_computed_file.pk, exception=e
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/smashing_utils.py:1040]: `exception=e` isn't necessary here as `logger.exception` will use the current stack traceback info to log the stacktrace and the exception.

File: workers/data_refinery_workers/processors/smashing_utils.py [hunk 2]
```diff
@@ -1026,16 +1026,39 @@ def write_tsv_json(job_context):
...
+        )
+        return "This sample's quant.sf file failed to download"
+
+    try:
+        with open(output_file_path, "r") as f:
+            header = f.readline()
+            # Filter out files that have been truncated and are missing their header
+            # https://salmon.readthedocs.io/en/latest/file_formats.html#quantification-file
+            if header.rstrip("\n") != "Name\tLength\tEffectiveLength\tTPM\tNumReads":
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/smashing_utils.py:1049]: Can you also include a link to the issue for this?

File: workers/data_refinery_workers/processors/smashing_utils.py (hunk 1 above)
Comment (wvauclain) [reply workers/data_refinery_workers/processors/smashing_utils.py:1040]: Oh sweet

File: workers/data_refinery_workers/processors/smashing_utils.py [hunk 3]
```diff
@@ -1064,10 +1087,17 @@ def sync_quant_files(output_path, samples: List[Sample], filtered_samples: Dict)
...
             # download this set of files, this will take a few seconds that should also help the db recover
-            executor.map(download_computed_file, sample_and_computed_files)
-            num_samples += len(sample_and_computed_files)
+            for error_reason in executor.map(download_quant_file, sample_and_computed_files):
+                if error_reason is not None:
+                    filtered_samples[sample.accession_code] = {
+                        **sample.to_metadata_dict(),
+                        "reason": error_reason,
+                    }
+                    continue
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/smashing_utils.py:1099]: I don't think this code is nested where you think it is. We loop over the samples in the `sample_page` and add them to `sample_and_computed_files`. Once we've gotten them all we map over the list and run `download_computed_files` on it. In this for loop, you're looping over every error reason from those downloads, and attributing them to `sample`, which is still defined as the last sample in `sample_page`.

I think you should instead be logging from `download_quant_file`. The rest of this doesn't really need to change I don't think.

Comment (wvauclain) [reply workers/data_refinery_workers/processors/smashing_utils.py:1099]: Good catch, I was originally doing more of the processing in this loop, but I missed that I removed where I updated sample.

Comment (kurtwheeler) [reply workers/data_refinery_workers/processors/smashing_utils.py:1099]: Oh wait, I forgot this is actually writing to the metadata file. In that case we do need to loop over it like this, however we need to have `download_quant_file` return the sample and the error message, so you can do `for (sample, error_reason) in executor.map(download_quant_file, sample_and_computed_files):`

**Pull request merged**